src/nvim/lua/executor.c

```diff
@@ -1523,6 +1523,21 @@ void nlua_do_ucmd(ucmd_T *cmd, exarg_T *eap)
   lua_pushstring(lstate, (const char *)eap->arg);
   lua_setfield(lstate, -2, "args");
 
+  // Split args by unescaped whitespace |<f-args>|
+  lua_newtable(lstate);
+  int length = (int)STRLEN(eap->arg);
+  int start = 0;
+  int end = 0;
+  int i = 1;
+  bool res = true;
+  while (res) {
+    res = uc_split_args_iter(eap->arg, i, &start, &end, length);
+    lua_pushlstring(lstate, (const char *)eap->arg + start, (size_t)(end - start + 1));
+    lua_rawseti(lstate, -2, i);
+    i++;
+  }
+  lua_setfield(lstate, -2, "fargs");
```

Comment:
Mmm I guess I'm just going with every time a command runs get the arguments split in a table. There's no current way to request `f-args` in the api, since It's rather a hack string replacement in the current implementation.

Do you think it should be requested/opt-in ?

Also if considered good, should we rename fargs to something else?

Reply:
You can’t request fargs when using a Lua callback. The whole table is passed to the callback function, and the function can use or not use any of the fields as it chooses.

I will take a closer look at this soon, but on first glance it looks good to me.

Reply:
Thanks for the review @lewis6991 👍🏽